modules/fedimint-dummy/src/lib.rs

```diff
@@ -46,19 +46,25 @@ pub struct DummyOutputConfirmation;
 #[derive(Debug, Clone)]
 pub struct DummyVerificationCache;
 
+#[derive(Debug)]
 pub struct DummyConfigGenerator;
 
 #[async_trait]
 impl ModuleGen for DummyConfigGenerator {
-    fn decoder(&self) -> DynDecoder {
-        DynDecoder::from_typed(DummyDecoder)
-    }
+    const KIND: ModuleKind = ModuleKind::from_static_str("dummy");
+    type Decoder = DummyDecoder;
 
-    fn module_kind(&self) -> ModuleKind {
-        const KIND: &str = "dummy";
-        ModuleKind::from_static_str(KIND)
+    fn decoder(&self) -> DummyDecoder {
+        DummyDecoder
     }
 
+    // TODO: is this method always required to be implemented? Or maybe have default impl that
```

Comment:
We should use the default. Or maybe even remove the default method and rellay on `Self::KIND` everywhere. Yes. That was my mistake. But maybe in another PR.